include/logger/loggerConfig.h

```diff
@@ -357,17 +358,19 @@ typedef struct _GPSConfig {
 #define DEFAULT_GPS_SATELLITE_CONFIG {"GPSSats", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 0, 0}
 #define DEFAULT_GPS_QUALITY_CONFIG {"GPSQual", "", 0, 5, DEFAULT_GPS_SAMPLE_RATE, 0, 0}
 #define DEFAULT_GPS_DOP_CONFIG {"GPSDOP", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 1, 0}
-
-#define DEFAULT_GPS_CONFIG {                   \
-		DEFAULT_GPS_LATITUDE_CONFIG,           \
-		DEFAULT_GPS_LONGITUDE_CONFIG,          \
-		DEFAULT_GPS_SPEED_CONFIG,              \
-		DEFAULT_GPS_DISTANCE_CONFIG,           \
-		DEFAULT_GPS_ALTITUDE_CONFIG,           \
-		DEFAULT_GPS_SATELLITE_CONFIG,          \
-		DEFAULT_GPS_QUALITY_CONFIG,            \
-		DEFAULT_GPS_DOP_CONFIG                 \
-         }
+#define DEFAULT_GPS_HEADING_CONFIG {"GPSHeading", "Degrees", 0, 360, DEFAULT_GPS_SAMPLE_RATE, 0, 0}
```

Comment:
My primary concern is removing "GPS" from the channel name, so it's
consistent with other channels that aren't GPS diagonstic info (Speed vs
GPSDOP).

According to that article, the magnetometer should be "Heading" and the GPS
calculated actual path should be called "Track".

I like "Compass" as it's more explicit as it describes an actual sensor,
consistent with the accelerometer and gyro channels.

Therefore - "Compass" and "Track"?

On Thu, Jul 30, 2015 at 8:02 AM, Andrew Stiegmann <notifications@github.com>
wrote:

> In include/logger/loggerConfig.h
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70#discussion_r35880071>
> :
>
> > @@ -357,17 +358,19 @@ typedef struct _GPSConfig {
> >  #define DEFAULT_GPS_SATELLITE_CONFIG {"GPSSats", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 0, 0}
> >  #define DEFAULT_GPS_QUALITY_CONFIG {"GPSQual", "", 0, 5, DEFAULT_GPS_SAMPLE_RATE, 0, 0}
> >  #define DEFAULT_GPS_DOP_CONFIG {"GPSDOP", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 1, 0}
> > -
> > -#define DEFAULT_GPS_CONFIG {                   \
> > -		DEFAULT_GPS_LATITUDE_CONFIG,           \
> > -		DEFAULT_GPS_LONGITUDE_CONFIG,          \
> > -		DEFAULT_GPS_SPEED_CONFIG,              \
> > -		DEFAULT_GPS_DISTANCE_CONFIG,           \
> > -		DEFAULT_GPS_ALTITUDE_CONFIG,           \
> > -		DEFAULT_GPS_SATELLITE_CONFIG,          \
> > -		DEFAULT_GPS_QUALITY_CONFIG,            \
> > -		DEFAULT_GPS_DOP_CONFIG                 \
> > -         }
> > +#define DEFAULT_GPS_HEADING_CONFIG {"GPSHeading", "Degrees", 0, 360, DEFAULT_GPS_SAMPLE_RATE, 0, 0}
>
> Not trying to argue with you... but for the sake of getting this right the
> first time:
> http://aviation.stackexchange.com/questions/8000/what-are-the-differences-between-bearing-vs-course-vs-direction-vs-heading-vs-tr
> … <#14edf7c7f71994a7_>
> On Thu, Jul 30, 2015 at 9:58 AM, Brent Picasso <notifications@github.com>
> wrote: In include/logger/loggerConfig.h <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70#discussion_r35879651>
> : > @@ -357,17 +358,19 @@ typedef struct _GPSConfig { > #define
> DEFAULT_GPS_SATELLITE_CONFIG {"GPSSats", "", 0, 20,
> DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define DEFAULT_GPS_QUALITY_CONFIG
> {"GPSQual", "", 0, 5, DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define
> DEFAULT_GPS_DOP_CONFIG {"GPSDOP", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 1, 0}
> > - > -#define DEFAULT_GPS_CONFIG { \ > - DEFAULT_GPS_LATITUDE_CONFIG, \ >
> - DEFAULT_GPS_LONGITUDE_CONFIG, \ > - DEFAULT_GPS_SPEED_CONFIG, \ > -
> DEFAULT_GPS_DISTANCE_CONFIG, \ > - DEFAULT_GPS_ALTITUDE_CONFIG, \ > -
> DEFAULT_GPS_SATELLITE_CONFIG, \ > - DEFAULT_GPS_QUALITY_CONFIG, \ > -
> DEFAULT_GPS_DOP_CONFIG \ > - } > +#define DEFAULT_GPS_HEADING_CONFIG
> {"GPSHeading", "Degrees", 0, 360, DEFAULT_GPS_SAMPLE_RATE, 0, 0} So the
> definition of Heading is direction or bearing - as in "I'm heading this
> way" A compass indicates the direction your're facing, regardless of your
> heading. People who understand how a compass works will understand the
> difference between the direction you're moving (Heading) and where you're
> pointing (Compass). So, I think these names are sufficiently clear. …
> <#14edf791b19a7159_> On Thu, Jul 30, 2015 at 7:52 AM, Andrew Stiegmann <
> notifications@github.com> wrote: In include/logger/loggerConfig.h <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70#discussion_r35878805>
> : > @@ -357,17 +358,19 @@ typedef struct _GPSConfig { > #define
> DEFAULT_GPS_SATELLITE_CONFIG {"GPSSats", "", 0, 20,
> DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define DEFAULT_GPS_QUALITY_CONFIG
> {"GPSQual", "", 0, 5, DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define
> DEFAULT_GPS_DOP_CONFIG {"GPSDOP", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 1, 0}
> > - > -#define DEFAULT_GPS_CONFIG { \ > - DEFAULT_GPS_LATITUDE_CONFIG, \ >
> - DEFAULT_GPS_LONGITUDE_CONFIG, \ > - DEFAULT_GPS_SPEED_CONFIG, \ > -
> DEFAULT_GPS_DISTANCE_CONFIG, \ > - DEFAULT_GPS_ALTITUDE_CONFIG, \ > -
> DEFAULT_GPS_SATELLITE_CONFIG, \ > - DEFAULT_GPS_QUALITY_CONFIG, \ > -
> DEFAULT_GPS_DOP_CONFIG \ > - } > +#define DEFAULT_GPS_HEADING_CONFIG
> {"GPSHeading", "Degrees", 0, 360, DEFAULT_GPS_SAMPLE_RATE, 0, 0} Say it a
> few times in your head and tell me if it sounds confusing if there is just
> "Heading" or "GPSHeading". If I rename it to "Heading" then people won't
> understand the difference between "Compass" and "Heading". I know I
> wouldn't get it. … <#14edf73653c4962e_> On Thu, Jul 30, 2015 at 9:46 AM,
> Brent Picasso <notifications@github.com> wrote: In
> include/logger/loggerConfig.h <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70#discussion_r35878131>
> : > @@ -357,17 +358,19 @@ typedef struct _GPSConfig { > #define
> DEFAULT_GPS_SATELLITE_CONFIG {"GPSSats", "", 0, 20,
> DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define DEFAULT_GPS_QUALITY_CONFIG
> {"GPSQual", "", 0, 5, DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define
> DEFAULT_GPS_DOP_CONFIG {"GPSDOP", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 1, 0}
> > - > -#define DEFAULT_GPS_CONFIG { \ > - DEFAULT_GPS_LATITUDE_CONFIG, \ >
> - DEFAULT_GPS_LONGITUDE_CONFIG, \ > - DEFAULT_GPS_SPEED_CONFIG, \ > -
> DEFAULT_GPS_DISTANCE_CONFIG, \ > - DEFAULT_GPS_ALTITUDE_CONFIG, \ > -
> DEFAULT_GPS_SATELLITE_CONFIG, \ > - DEFAULT_GPS_QUALITY_CONFIG, \ > -
> DEFAULT_GPS_DOP_CONFIG \ > - } > +#define DEFAULT_GPS_HEADING_CONFIG
> {"GPSHeading", "Degrees", 0, 360, DEFAULT_GPS_SAMPLE_RATE, 0, 0} Compass
> channel is called "Compass" Brent Picasso Autosport Labs Technology for
> Race and Street … <#14edf6e4878fef38_> On Jul 30, 2015 7:45 AM, "Andrew
> Stiegmann" <notifications@github.com> wrote: In
> include/logger/loggerConfig.h <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70#discussion_r35877981>
> : > @@ -357,17 +358,19 @@ typedef struct _GPSConfig { > #define
> DEFAULT_GPS_SATELLITE_CONFIG {"GPSSats", "", 0, 20,
> DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define DEFAULT_GPS_QUALITY_CONFIG
> {"GPSQual", "", 0, 5, DEFAULT_GPS_SAMPLE_RATE, 0, 0} > #define
> DEFAULT_GPS_DOP_CONFIG {"GPSDOP", "", 0, 20, DEFAULT_GPS_SAMPLE_RATE, 1, 0}
> > - > -#define DEFAULT_GPS_CONFIG { \ > - DEFAULT_GPS_LATITUDE_CONFIG, \ >
> - DEFAULT_GPS_LONGITUDE_CONFIG, \ > - DEFAULT_GPS_SPEED_CONFIG, \ > -
> DEFAULT_GPS_DISTANCE_CONFIG, \ > - DEFAULT_GPS_ALTITUDE_CONFIG, \ > -
> DEFAULT_GPS_SATELLITE_CONFIG, \ > - DEFAULT_GPS_QUALITY_CONFIG, \ > -
> DEFAULT_GPS_DOP_CONFIG \ > - } > +#define DEFAULT_GPS_HEADING_CONFIG
> {"GPSHeading", "Degrees", 0, 360, DEFAULT_GPS_SAMPLE_RATE, 0, 0} How do you
> intend to differentiate between compass heading and GPS heading? — Reply to
> this email directly or view it on GitHub <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70/files#r35877981>
> . — Reply to this email directly or view it on GitHub <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70/files#r35878131>
> . -- — Reply to this email directly or view it on GitHub <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70/files#r35878805>
> . -- Brent Picasso *Autosport Labs <http://www.autosportlabs.com>*
> *Technology for Race and Street* — Reply to this email directly or view it
> on GitHub <
> https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70/files#r35879651>
> .
> --
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/70/files#r35880071>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*